docs/UserGuide.md

```diff
@@ -126,25 +134,43 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
 Examples:
 * `find John` returns `john` and `John Doe`
 * `find alex david` returns `Alex Yeoh`, `David Li`<br>
-  ![result for 'find alex david'](images/findAlexDavidResult.png)
+  <!--- ![result for 'find alex david'](images/findAlexDavidResult.png) --->
+
+### Filtering residents by field : `filter`
+
+Shows a list of residents whose fields match the input keywords.
+
+Format: `filter [/specifier] KEY:VALUE [ADDITIONAL_KEYS:ADDITIONAL VALUES]`
+
+* The default specifier, if not specified by the user is `/all`.
+* Currently, only two specifiers are supported:
+  * `/all` returns a resident if it fulfills **all** of the specified keywords (Logical AND operation).
+  * `/any` returns a resident if it fulfills **any** of the specified keywords (Logical OR operation).
+* Repeated keys are not permitted for `/all` specifier, e.g. `filter /all house:Draco house:Aquila` will not work.
+* Valid keys are those included when [adding](#adding-a-resident--add) or [editing](#editing-an-existing-resident--edit) a resident e.g. `NAME, PHONE_NUMBER, EMAIL, FLOOR, UNIT, and TAG`
+
+Examples:
+* `filter /all house:Draco gender:Male` returns residents who are in Draco house, **and** are Male.
+* `filter /any house:Draco house:Aquila` returns residents belonging to either `Draco` **or** `Aquila` house.
+* `filter major:ComputerScience` returns residents who are in ComputerScience
 
-### Deleting a person : `delete`
+### Deleting a resident : `delete`
 
-Deletes the specified person from the address book.
+Deletes the specified resident from the RC4HDB database.
 
 Format: `delete INDEX`
 
 * Deletes the person at the specified `INDEX`.
-* The index refers to the index number shown in the displayed person list.
+* The index refers to the index number shown in the displayed resident list.
 * The index **must be a positive integer** 1, 2, 3, …​
```

Comment:
* Deletes the person with the specified `MATRIC_NO`.

Reply:
Remove the 2 points about `INDEX`.

Reply:
Let's wait for team input.

Reply:
Let's revert back to using INDEX, this change should no longer be necessary :)

Reply:
`INDEX` will continue to be used for now, maybe as a future feature, we will overload delete to make use of `MATRIC_NO`.